crypto/property/property_parse.c

```diff
@@ -453,6 +453,33 @@ int ossl_property_has_optional(const OSSL_PROPERTY_LIST *query)
     return query->has_optional ? 1 : 0;
 }
 
+int ossl_property_is_enabled(OPENSSL_CTX *ctx,  const char *property_name,
+                             const OSSL_PROPERTY_LIST *prop_list)
+{
+    int i;
+    OSSL_PROPERTY_IDX name_id;
+    const PROPERTY_DEFINITION *prop = NULL;
+
+    if (prop_list == NULL)
+        return 0;
+
+    if (!parse_name(ctx, &property_name, 0, &name_id))
+        return 0;
+
+    prop = prop_list->properties;
+    for (i = 0; i < prop_list->n; ++i) {
```

Comment:
In crypto/serializer/serializer_pkey.c, there's a very simple parameter check, using `ossl_parse_query` and `ossl_property_match_count`. Why do you need a specially coded loop to do something similar?

Reply:
I tried this initially and it failed.. match_count returns 1 for optionals..

Reply:
If "?fips=yes" is in the default query, `ossl_property_match_count` will return 1 but FIPS mode isn't mandated.

Additionally, the matching code isn't designed to match a query against a query -- there are cases where it won't work sensibly.  I should have used two named types PROPERTY_DEFINITION_LIST and PROPERTY_QUERY_LIST to highlight the difference.

Reply:
Ah, gotcha. In the serializer case, I'm matching against a property definition.
OK then.